### PR TITLE
Constrained Cartesian planning using moveit commander

### PR DESCRIPTION
**Pull request by srsidd**
### Description
The `moveit::planning_interface::MoveGroup::computeCartesianPath` takes in an optional argument for path constraints, but the wrapper for using `move_group` in python using `moveit_commander` does not have the same option. 
This PR adds that feature to allow for constrained cartesian planning using `moveit_commander`.

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [x] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by mlautman** [comment]
Glad to see the python interfaces getting some love. Would you mind adding a python script to the [moveit_commander/demo directory](https://github.com/ros-planning/moveit/tree/kinetic-devel/moveit_commander/demos) showing how to use path constraints in python?

**Comment by mlautman** [comment]
For context, I'm currently re-vamping the tutorials and would like to include this feature in the moveit_commander tutorial. Also, having a demo script in that folder will help us quickly test that the changes you made work as expected.

File: moveit_commander/demos/plan_with_constraints.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2013, Willow Garage, Inc.
```
Comment (davetcoleman) [inline moveit_commander/demos/plan_with_constraints.py]: You're welcome to change this to HMI

File: moveit_commander/demos/plan_with_constraints.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    rospy.init_node('moveit_py_demo', anonymous=True)
+
+    robot = RobotCommander()
+    rospy.sleep(1)
+
+    # Plan to a random location
+    a = robot.right_arm
+    a.set_start_state(RobotState())
+
+    print "cutternt pose:"
```
Comment (davetcoleman) [inline moveit_commander/demos/plan_with_constraints.py]: "current"

File: moveit_commander/demos/plan_with_constraints.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    # Move down
+    wpose.position.z -= 0.10
+    waypoints.append(wpose)
+
+    # Move to the side
+    wpose.position.y += 0.05
+    waypoints.append(wpose)
+
+    plan, fraction = a.compute_cartesian_path(waypoints, 0.01, 0.0, path_constraints=c)
+    print 'Plan success rate: ', fraction
```
Comment (davetcoleman) [inline moveit_commander/demos/plan_with_constraints.py]: "Percent plan complete:"
not really a rate as its run just once

**Pull request merged**